pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/validator/ModelValidatorImpl.groovy

```diff
@@ -343,16 +342,21 @@ class ModelValidatorImpl implements ModelValidator {
                     List<DescribableParameter> requiredParams = model.parameters.findAll { it.isRequired() }
 
                     if (requiredParams.size() != meth.args.size()) {
-                        errorCollector.error(meth, Messages.ModelValidatorImpl_WrongNumberOfStepParameters(meth.name, requiredParams.size(), meth.args.size()))
-                        valid = false
+                        // TODO: This is a specialized hack for allowing single-required-Boolean-parameter constructors
+                        // to be called like "foo()", Groovy-style, passing null as the parameter value.
+                        if (!(requiredParams.size() == 1 &&
+                            meth.args.isEmpty() &&
+                            requiredParams.get(0).erasedType == Boolean.class)) {
+                            errorCollector.error(meth, Messages.ModelValidatorImpl_WrongNumberOfStepParameters(meth.name, requiredParams.size(), meth.args.size()))
+                            valid = false
+                        }
                     } else {
                         requiredParams.eachWithIndex { DescribableParameter entry, int i ->
                             def argVal = meth.args.get(i)
                             if (argVal instanceof ModelASTMethodCall) {
                                 valid = validateElement((ModelASTMethodCall) argVal)
                             } else {
                                 if (!validateParameterType((ModelASTValue) argVal, entry.erasedType)) {
-                                    errorCollector.error((ModelASTValue)argVal, Messages.ModelValidatorImpl_WrongParameterType(entry.name))
```

Comment:
Because it was redundant - we are already reporting errors in `validateParameterType`.